Sources/Public/iOS/BundleImageProvider.swift

```diff
@@ -23,15 +23,30 @@ public class BundleImageProvider: AnimationImageProvider {
   ///
   /// - Parameter bundle: The bundle containing images for the provider.
   /// - Parameter searchPath: The subpath is a path within the bundle to search for image assets.
+  /// - Parameter cacheEnabled: The flag to turn on image cache to reuse `CGImage` for asset
   ///
-  public init(bundle: Bundle, searchPath: String?) {
+  public init(bundle: Bundle, searchPath: String?, cacheEnabled: Bool = true) {
     self.bundle = bundle
     self.searchPath = searchPath
+    imageCache = cacheEnabled ? .init() : nil
   }
 
   // MARK: Public
 
   public func imageForAsset(asset: ImageAsset) -> CGImage? {
+    if let imageCache = imageCache, let cachedImage = imageCache.object(forKey: asset.id as NSString) {
```

Comment:
Is it safe to cache assets between different animations? Couldn’t two animations use the same id for different assets?

Reply:
Good point! We use `asset.name` and `asset.directory` below when retrieving the image (and _not_ the `id`), so we should use those as the cache key.

Reply:
The provider is created for each AnimationView https://github.com/airbnb/lottie-ios/blob/master/Sources/Public/Animation/AnimationView.swift#L97
Thus, the asset id should be unique in an lottie json I think.
The `asset.name` may represent for base64 image data, its size is too big. @marcelofabri @calda

Reply:
> Is it safe to cache assets between different animations? Couldn’t two animations use the same id for different assets?

This inspire me that if it is possible we could share `BundleImageProvider` between `AnimationView`s. If the provider is shared, the memory usage of image would be reduced if an animation is used in more than on `AnimationView`. Then, the cache key may need to have the animation json hash as key. @marcelofabri

Reply:
An individual `AnimationView` can display multiple different Lottie animations over time

Reply:
I think we need a solution for using `name` and `directory` in the cache key. Otherwise it will be possible for the cache to return an incorrect result.

> The asset.name may represent for base64 image data, its size is too big.

Too big for what? Does `NSCache` not allow you to store these?

Reply:
It may cost cpu for calculate hash value of big string (` name`) as NSCache key @calda

Reply:
> An individual `AnimationView` can display multiple different Lottie animations over time

I would like to see if we could reset cache for when animation changed.

Reply:
I updated the PR with different solution, the cached provider is recreated for each time `Animation` updated.
This solves the cache key conflict problem among different Lottie animations. @calda

Reply:
Sounds good, it seems safe to assume that an individual animation can only have a single image with a given `id`.